packages/langium/src/utils/cst-util.ts

```diff
@@ -74,6 +74,36 @@ export function toDocumentSegment(node?: CstNode): DocumentSegment | undefined {
     };
 }
 
+export enum RangeComparison {
+    Before = 0,
+    After = 1,
+    OverlapFront = 2,
+    Inside = 3,
```

Comment:
I'd put this to the last position because a partial overlap "is less than" a full overlap.